### PR TITLE
[Scope change] add warning dialog to create

### PR DESCRIPTION
**Pull request by Gustav-Eikaas**
# Description

Please include a summary of the change and which issue is fixed.
List any external dependencies that are required for this change.

Completes:
[AB#212723](https://dev.azure.com/Equinor/fa63ceea-8883-41e2-8823-a3b54e4be178/_workitems/edit/212723)

## Checklist:

-   [x] I have performed a self-review of my own code.
-   [ ] I have linked my DevOps task using the AB# tag.
-   [x] My code is easy to read, and comments are added where needed.
-   [ ] My code is covered by tests.

### CODE REVIEW

File: src/apps/ScopeChangeRequest/Components/Form/ScopeChangeRequestForm.tsx [hunk 1]
```diff
@@ -29,18 +32,24 @@ interface ScopeChangeRequestFormProps {
 
 export const ScopeChangeRequestForm = ({
     closeScrim,
+    setHasUnsavedChanges,
 }: ScopeChangeRequestFormProps): JSX.Element => {
     const { handleInput, isValid, state } = useScopeChangeFormState();
     const { createScopeChangeMutation } = useRequestMutations();
     const queryClient = useQueryClient();
 
+    const handleChange = (key: keyof ScopeChangeFormModel, value: unknown) => {
```
Comment (Noggling) [inline src/apps/ScopeChangeRequest/Components/Form/ScopeChangeRequestForm.tsx:41]: value unknown?

Comment (Noggling) [reply src/apps/ScopeChangeRequest/Components/Form/ScopeChangeRequestForm.tsx:41]: vil ikke dette være 

```TS
function <K extends keyof ScopeChangeFormModel>(key: K, value:ScopeChangeFormModel[K] )
```

eller noe i den duren ?

Comment (Gustav-Eikaas) [reply src/apps/ScopeChangeRequest/Components/Form/ScopeChangeRequestForm.tsx:41]: Ja og nei, kan type det opp sånn men ender bare opp med masse casting for union types osv..

**Review by Noggling — APPROVED** [review]
ok

**Pull request merged**